common-lib/src/cmr/common/generics.clj

```diff
@@ -117,3 +119,116 @@
+  Parameters:
+  * file-name: [ingest | search]
+  Returns: string"
+  [file-name]
+  (string/join (seq (for [[k,v] (latest-approved-documents)]
+                      (read-generic-doc-file file-name k (str v))))))
+
+(def search-depth-map (hash-map 3 0 4 4 5 8))
+(def ingest-depth-map (hash-map 3 8))
+
+(defn get-toc-headers-from-markdown
+  "From the markdown file string of either the ingest or search,
+  dyanamically create the table of content
+  Parameters:
+   * generic-markdown: markdown file for a generic's ingest.md or search.md
+   Returns: string"
+  [generic-markdown]
+  (re-seq #"\#+\s+<a\s.*" generic-markdown))
+
+(defn fill-in-generic-name
+  "Return the html for the table of contents given a generic concept
+   Parameters:
+   * table-of-contents-template: [ingest-table-of-contents-template | search-table-of-contents-template]
+   * generic-type: grid, dataqualitysummary etc
+   Returns: string"
+  [table-of-contents-template generic-type]
+  (-> table-of-contents-template
+      (string/replace #"%generic%" generic-type)
+      (string/replace #"%uc-generic%" (string/capitalize generic-type))
+      (string/replace #"%plural-generic%" (inf/plural generic-type))
+      (string/replace #"%uc-plural-generic%" (inf/plural (string/capitalize generic-type)))))
+
+;For each of the depth in the map get the output
+(defn build-markdown-toc
+  "Get the markdown file for a generic based on the api string being read in
+   Depth is a measure of how many spaces we need from the start of the line for the
+   markdown
+   Parameters:
+   * depth: integer a count of the number of spaces
+   * content: string, the content for the toc
+   * link: string, the link to the location in the document"
+  [depth content link]
+  (str (apply str (repeat depth " ")) (format "* [%s](#%s)\n" link content)))
+
+(defn get-toc-data
+  "Parses out the toc information from the api.md to build the markdown for the toc
+   Parameters:
+   * table-of-content-headers: string list, the headers in the document
+   * doc-type: string [ingest | search]"
+  ([table-of-content-headers doc-type]
+   (let [depth (if (= doc-type "ingest") (ingest-depth-map (count (re-seq #"#" table-of-content-headers))) (search-depth-map (count (re-seq #"#" table-of-content-headers))))
+         link (peek (re-find #"=\"(.*)\">" table-of-content-headers))
+         content (peek (re-find #">\s+([^<]*)$" table-of-content-headers))]
+     (build-markdown-toc depth link content))))
+
+(defn format-generic-toc
+  "Return the specific schema's documentation files given the schema keyword name and version number.
+   if the file cannot be read, return an empty string which will have no impact on the API document.
+   Parameters:
+   * file-name: [ingest | search]
+   * generic-keyword: [:grid | ...]
+   * generic-version: 0.0.1
+   Returns: string"
+  [file-name generic-keyword generic-version]
+  (try
+    (-> "schemas/%s/v%s/%s.md"
```

Comment:
wait, why not use the function in common to load this file?

Reply:
re-factored to use the read-generic-doc function